_source/_posts/2022-08-19-build-crud-spring-and-vue.md

```diff
@@ -0,0 +1,1057 @@
+---
+layout: blog_post
+title: "Build a Simple CRUD App with Spring Boot and Vue.js"
+author: andrew-hughes
+by: contractor
+communities: [java,javascript]
+description: "Create a CRUD (create, read, update, and delete) application using Spring Boot and Vue.js."
+tags: [java, spring-boot, vue, javascript]
+tweets:
+- ""
+- ""
+- ""
+image:
+type: conversion
+github: https://github.com/oktadev/okta-spring-boot-vue-crud-example
+---
+
+You're going to use Vue and Spring Boot to build a todo list web application. The application will include CRUD abilities, meaning that you will be able to **c**reate, **r**ead, **u**pdate, and **d**elete the todo items on the Spring Boot API via the client. The Vue frontend client will use the Quasar framework for presentation. Both the Spring Boot API and the Vue client will be secured with OAuth 2.0 and OpenID Connect (OIDC) using Okta as the security provider.
+
+{% img blog/spring-boot-crud-update/spring-and-vue.png alt:"Spring Boot, Vue, and Okta logos" width:"500" %}{: .center-image }
+
+This project has two major parts:
+
+- Spring Boot API
+- Vue client
+
+The Spring Boot app will include an H2 in-memory database and will use Spring Data JPA to map our todo data model to a database table for persistence. As you'll see, the server will leverage Spring Boot's ability to quickly expose data via a REST API with a minimum of configuration.
+
+The client will use [Vue 3](https://vuejs.org/) and the Quasar framework. [The Quasar framework](https://quasar.dev/) provides components and layout tools to help build Vue applications quickly with a consistent, high-quality user interface.
+
+**Prerequisites:**
+
+- [Java 11](https://adoptium.net/): or use [SDKMAN!](https://sdkman.io/) to manage and install multiple versions
+- [Okta CLI](https://cli.okta.com/manual/#installation)
+- [HTTPie](https://httpie.org/doc#installation)
+- [Node 16+](https://nodejs.org)
+- [Vue CLI](https://cli.vuejs.org/guide/installation.html)
+
+You will need a free Okta Developer account if you don't already have one. But you can wait until later in the tutorial and use the Okta CLI to log in or register for a new account.
+
+{% include toc.md %}
+
+## Create Okta OIDC app
+
+Open a Bash shell. Create a parent directory for the project. Eventually, this will include both the resource server and client projects.
+
+```bash
+mkdir spring-boot-vue-crud
+cd spring-boot-vue-crud
+```
+
+{% include setup/cli.md type="spa" framework="Vue" loginRedirectUri="http://localhost:8080/callback" %}
+
+Your console should finish with something like this:
+
+```bash
+...
+Configuring a new OIDC Application, almost done:
+Created OIDC application, client-id: 0oa5rdbh01J0293u209
+|
+Okta application configuration: 
+Issuer:    https://dev-123567.okta.com/oauth2/default
+Client ID: 0oa5rdbh01J0293u209
+```
```

Comment:
If you look at the rendered preview of this post, you'll see this same information under the "What does the Okta CLI do?" expandable section. For that reason, you might want to remove this and just say something like "The client ID and issuer URI will be printed to your terminal. Copy them somewhere safe."